compiler/rustc_ast_lowering/src/lifetime_collector.rs

```diff
@@ -38,32 +45,39 @@ impl<'this, 'ast: 'this> Visitor<'ast> for LifetimeCollectVisitor<'this, 'ast> {
     }
 
     fn visit_ty(&mut self, t: &'ast Ty) {
-        if let TyKind::BareFn(_) = t.kind {
-            self.current_binders.push(t.id);
-        }
-        visit::walk_ty(self, t);
-        if let TyKind::BareFn(_) = t.kind {
-            self.current_binders.pop();
+        match t.kind {
```

Comment:
Why not check lifetime_res(t.id) in the TyKind::Rptr(None, _) case?
You'll need to check it when visiting paths too.

Reply:
by check lifetime_res(t.id) you meant to not push to underscore_lifetimes if the resolution's binder is already included?

this ...

```rust
            TyKind::Rptr(None, _) => {
                let res = self.resolver.get_lifetime_res(t.id).unwrap_or(LifetimeRes::Error);

                if let Some(binder) = res.binder() && !self.current_binders.contains(&binder) {
                    if !self.collected_lifetimes.contains(&lifetime) {
                        self.underscore_lifetimes.push((t.id, t.span));
                    }
                }
                visit::walk_ty(self, t);
            }
```